dev/bus/tests/serial_unit.cpp

```diff
@@ -0,0 +1,52 @@
+#include <CppUTest/TestHarness.h>
+#include <CppUTest/CommandLineTestRunner.h>
+#include <CppUTestExt/MockSupport.h>
+
+#include "dev/serial.hpp"
+#include "mocks/platform_bus.hpp"
+
+// Error code helper
+// TODO: move it to 'utils' headers and protect with check of
+// current test state (enabled or disabled)
+static SimpleString StringFrom(ecl::err err)
+{
+    return SimpleString{ecl::err_to_str(err)};
+}
+
+using serial_t = ecl::serial<platform_mock>;
+
+TEST_GROUP(serial)
+{
+    void setup()
+    {
+    }
+
+    void teardown()
+    {
+        mock().clear();
+    }
+};
+
+
+TEST(serial, init)
+{
```

Comment:
Let's be a more TDD'ish - add more tests, specifically at corner cases when overflow will occur